pkg/oci/load_balancer_spec.go

```diff
@@ -156,27 +169,26 @@ func NewLBSpec(svc *v1.Service, nodes []*v1.Node, defaultSubnets []string, sslCf
 }
 
 // Certificates builds a map of required SSL certificates.
-func (s *LBSpec) Certificates() (map[string]loadbalancer.CertificateDetails, error) {
```

Comment:
This was explicitly designed like this so that we didn't Get() the SSL secret every time that EnsureLoadbalancer is called. I'm not clear that behaviour has been maintained?

Reply:
Yes, it's almost the same, the call  to this function is maintained, the difference is that in the create load balancer stage, we create a map of all the certificates the need to be created. So we declare a certificates map in the createLB method, and then, for each resource type (listener and backendset) we read the ssl certificate details and add them the map

Reply:
I guess I could have returned a certificate for each type and then created the map from there, makes more sense